src/normalizeOptions.js

```diff
@@ -51,29 +52,46 @@ function normalizeRoot(opts) {
   }
 }
 
-function normalizeAlias(opts) {
-  opts.regExps = [];
+function getAliasPair(key, value) {
+  const parts = value.split('\\\\');
+
+  function substitute(execResult) {
+    return parts
+      .map(part =>
+        part.replace(/\\\d+/g, number => execResult[number.slice(1)] || ''),
+      )
+      .join('\\');
+  }
+
+  return [new RegExp(key), substitute];
+}
 
+function normalizeAlias(opts) {
   if (opts.alias) {
-    Object.keys(opts.alias)
-      .filter(isRegExp)
-      .forEach((key) => {
-        const parts = opts.alias[key].split('\\\\');
-
-        function substitute(execResult) {
-          return parts
-            .map(part =>
-              part.replace(/\\\d+/g, number => execResult[number.slice(1)] || ''),
-            )
-            .join('\\');
+    const { alias } = opts;
+    const aliasKeys = Object.keys(alias);
+    let warnedAboutNpmPrefix = false;
+
+    const nonRegExpAliases = aliasKeys
+      .filter(key => !isRegExp(key))
+      .map((key) => {
+        let value = alias[key];
+
+        if (value.startsWith('npm:') && !warnedAboutNpmPrefix) {
```

Comment:
It will be a 3.0 so no problem removing the deprecated things :) Especially since there's a way to do it without using the prefix.